### PR TITLE
feat: add cli feature flags documentation

### PR DESCRIPTION
**Pull request by attilah**
*Issue #, if available:*

N/A

*Description of changes:*

This PR adds feature flag documentation of the CLI.

Feature flag documentation rendering is done by `amplify-feature-flag*` components. Feature flags must be managed in the `~/client/src/amplify-ui/cli-feature-flag/feature-flags.json` file.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: client/src/amplify-ui/cli-feature-flag/feature-flag-summary.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import {Component, Host, h, Prop, Element} from "@stencil/core";
+import {FeatureFlag} from "./feature-flag";
+import {tableStyle, summaryCell} from "./feature-flag.style";
+
+@Component({tag: "amplify-feature-flag-summary", shadow: false})
+export class AmplifyFeatureFlagSummary {
+  @Element() el: HTMLElement;
```
Comment (jordanranz) [inline client/src/amplify-ui/cli-feature-flag/feature-flag-summary.tsx:7]: I do not see this used anywhere, is it needed?

**Comment by jordanranz** [comment]
Could you provide a screenshot of this on mobile?

File: client/src/amplify-ui/cli-feature-flag/feature-flag-summary.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        {this.feature.description ? (
+          <p>{this.feature.description}</p>
+        ) : (
+          undefined
+        )}
+        <table class={tableStyle}>
+          <thead>
+            <tr>
+              <th class={summaryCell}>Type</th>
```
Comment (jordanranz) [inline client/src/amplify-ui/cli-feature-flag/feature-flag-summary.tsx:30]: You could improve this CSS specification by adding a single class to `tr` and adding a rule for `th` elements to implement the style specified in `summaryCell`

File: client/src/amplify-ui/cli-feature-flag/feature-flag.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  removalDate?: string;
+  values: Value[];
+};
+
+export type Value = {
+  value: string;
+  description: string;
+  defaultNewProject: boolean;
+  defaultExistingProject: boolean;
+};
```
Comment (jordanranz) [inline client/src/amplify-ui/cli-feature-flag/feature-flag.tsx:29]: Can we move these into a `.types.ts` file similar to: https://github.com/aws-amplify/docs/blob/master/client/src/docs-ui/page/page.types.ts

File: client/src/amplify-ui/cli-feature-flag/feature-flag.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+@Component({tag: "amplify-feature-flag", shadow: false})
```
Comment (jordanranz) [inline client/src/amplify-ui/cli-feature-flag/feature-flag.tsx:31]: NIT: This seems to be representing multiple feature flags. Should this be plural? or `amplify-feature-flag-section`.

File: client/src/amplify-ui/cli-feature-flag/feature-flag-summary.tsx (hunk 1 above)
Comment (attilah) [reply client/src/amplify-ui/cli-feature-flag/feature-flag-summary.tsx:7]: Nope, leftover

File: client/src/amplify-ui/cli-feature-flag/feature-flag.tsx (hunk 4 above)
Comment (attilah) [reply client/src/amplify-ui/cli-feature-flag/feature-flag.tsx:31]: pluralized.

**Pull request merged**